storage/src/main/java/tech/pegasys/teku/storage/client/RecentChainData.java

```diff
@@ -530,6 +534,19 @@ public UInt64 getLatestValidFinalizedSlot() {
     return store == null ? UInt64.ZERO : store.getLatestValidFinalizedSlot();
   }
 
+  public boolean isOptimisticSyncPossible() {
+    return store != null
+        && !store.getLatestFinalized().getExecutionBlockHash().map(Bytes32::isZero).orElse(true);
+  }
+
+  public boolean enqueueExecutionPayloadExecutionRetry(SignedBeaconBlock block) {
+    return payloadExecutionRetryQueue.add(block);
```

Comment:
We are going to have to have some kind of limit on the number of blocks we hold in this queue, otherwise it will fairly quickly cause an OOM error.  For other cases of pending attestations and blocks we use `PendingPool` though I'm not sure if it will be applicable here.

I suspect we should only hold a fairly small number of blocks from the same fork - if we're unable to import blocks for any length of time we can drop the later blocks as we'll wind up having to go into long sync mode and refetch them from the network anyway.  Plus the `BlockManager` already knows how to backfill blocks when the parent is unknown.

In fact, I suspect if we get this right our pending execution queue should only ever contain one block from a fork (ie a block in this list should never also have it's parent in the list).  If we haven't imported the parent, we should just return `BlockImportResult.FAILED_UNKNOWN_PARENT` from `ForkChoice.onBlock` and `BlockManager` already handles that.  So the only blocks in this list should be ones that pass all the state transition checks except we were unable to verify their payload.